library/alloc/tests/str.rs

```diff
@@ -1921,3 +1921,11 @@ fn different_str_pattern_forwarding_lifetimes() {
 
     foo::<&str>("x");
 }
+
+#[test]
+fn test_str_concat() {
+    let a: String = "hello".to_string();
+    let b: String = "world".to_string();
+    let s: String = format!("{}{}", a, b);
+    assert_eq!(s.as_bytes()[9], 'd' as u8);
+}
```

Comment:
This should go in `string.rs` instead I think. r=me after that

Reply:
@jyn514 Shall I move in `string.rs` or wait for @matklad to provide any feedback?

Reply:
Go ahead and move it.

Reply:
Done @jyn514 . Please check and approve